rpcs/httpTxSync.go

```diff
@@ -103,19 +102,18 @@ func (hts *HTTPTxSync) Sync(ctx context.Context, bloom *bloom.Filter) (txgroups
 	if !ok {
 		return nil, fmt.Errorf("cannot HTTPTxSync non http peer %T %#v", peer, peer)
 	}
+	var syncURL string
 	hts.rootURL = hpeer.GetAddress()
+
 	client := hpeer.GetHTTPClient()
 	if client == nil {
-		client = &http.Client{}
-		client.Transport = hts.peers.GetRoundTripper()
-	}
-	parsedURL, err := network.ParseHostOrURL(hts.rootURL)
-	if err != nil {
-		hts.log.Warnf("txSync bad url %v: %s", hts.rootURL, err)
-		return nil, err
+		client, err = hts.peers.GetHTTPClient(hts.rootURL)
```

Comment:
Have we run this in hybrid mode already/feel good it works under either protocol (sanity check me if tx sync is only on legacy)?

Reply:
`TestTxSync` (see this sub pr for more narrowed view https://github.com/algorand/go-algorand/pull/5922) checks tx sync over p2p net.
The code is expected to work with any kind of peers (it takes one random to try)